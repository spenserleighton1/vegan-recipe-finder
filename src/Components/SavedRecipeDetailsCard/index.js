import React, { Component } from 'react';
import PropTypes from 'prop-types';

class SavedRecipeCard extends Component {
  constructor() {
    super();
    this.state = { displayDetails: false };
  }

  handleClick() {
    this.setState({ displayDetails: !this.state.displayDetails });
  }

  render() {
    const { title, image, ingredients, linkUrl } = this.props;
    const details = this.state.displayDetails ? 
      <ul>
        { ingredients.map((ingredient, index) => <li key={ index }>{ ingredient }</li>)}
        <li><a href={ linkUrl }>Directions</a></li>
      </ul> : '';
    return (
      <div className='recipe-card'>
        <h3 className='recipe-title'>{ title }</h3>
        <img className='recipe-img' src={ image } alt={ title } />
        <button onClick={() => { this.handleClick(); } }> Details </button>;
        { details }
      </div>
    );
  }
}

export default SavedRecipeCard;
      
SavedRecipeCard.propTypes = {
  title: PropTypes.string,
  image: PropTypes.url,
  ingredients: PropTypes.array,
  linkUrl: PropTypes.url
};