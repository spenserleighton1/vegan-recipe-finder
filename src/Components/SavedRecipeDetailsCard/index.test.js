import React from 'react';
import { shallow } from 'enzyme';
import SavedRecipeDetailsCard from './'

describe('SavedRecipeDetailsCard', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SavedRecipeDetailsCard />)

    expect(wrapper).toMatchSnapshot()
  })
})