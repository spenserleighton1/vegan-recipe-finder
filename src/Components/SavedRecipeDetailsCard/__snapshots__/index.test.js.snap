// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SavedRecipeDetailsCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SavedRecipeCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="recipe-title"
/>,
        <img
          alt={undefined}
          className="recipe-img"
          src={undefined}
/>,
        <button
          onClick={[Function]}
>
           Details 
</button>,
        "",
      ],
      "className": "recipe-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "recipe-title",
        },
        "ref": null,
        "rendered": null,
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "recipe-img",
          "src": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Details ",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": " Details ",
        "type": "button",
      },
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="recipe-title"
/>,
          <img
            alt={undefined}
            className="recipe-img"
            src={undefined}
/>,
          <button
            onClick={[Function]}
>
             Details 
</button>,
          "",
        ],
        "className": "recipe-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "recipe-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "recipe-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Details ",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": " Details ",
          "type": "button",
        },
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
